### PR TITLE
Memzero secrets

### PR DESCRIPTION
**Pull request by nmahendru**

### CODE REVIEW

File: core/src/sign.c [hunk 1]
```diff
@@ -480,30 +481,39 @@ compute_output_hash(char xpub[static MULTISIG_PARTS][XPUB_SIZE],
...
+ * Confidential items have been marked.
+ * An effort has been made to zeroize them regardless of success or failure of this function.
+ *
  */
 Result handle_sign_tx(InternalCommandRequest_SignTxRequest *request,
                       InternalCommandResponse_SignTxResponse *response) {
+  Result r = Result_SUCCESS;
+  uint8_t CONFIDENTIAL seed[SHA512_DIGEST_LENGTH] = {0};
+  HDNode CONFIDENTIAL wallet;
+  memzero((void *) &wallet, sizeof(HDNode));
```
Comment (syncom) [inline core/src/sign.c:494]: I believe casting the pointer to `void *` in this case (and cases later in the file) is unnecessary, and strictly speaking incorrect. In the declaration of `memset`, the parameter type `void *` means it takes a generic pointer (any pointer type) as an argument.

File: core/src/sign.c [hunk 2]
```diff
@@ -480,30 +481,39 @@ compute_output_hash(char xpub[static MULTISIG_PARTS][XPUB_SIZE],
...
+  char CONFIDENTIAL xpub[MULTISIG_PARTS][XPUB_SIZE] = {0};
+  uint8_t CONFIDENTIAL public_key[33] = {0};
```
Comment (oreparaz) [inline core/src/sign.c:496]: any special reason for moving the declaration of `public_key` up here? the nice thing of declaring it later is that the scope is smaller (making it harder to introduce bugs).

File: core/src/sign.c [hunk 3]
```diff
@@ -613,6 +623,14 @@ Result handle_sign_tx(InternalCommandRequest_SignTxRequest *request,
     memcpy(response->signatures[i].hash, hash, sizeof(hash));
   }
   response->signatures_count = request->inputs_count;
+  cleanup:
+  // cleanup the house be it error or success.
+  memzero((void *) seed, SHA512_DIGEST_LENGTH);
+  memzero((void *) &wallet, sizeof(HDNode));
+  for (int i = 0 ; i < MULTISIG_PARTS ; i++){
```
Comment (oreparaz) [inline core/src/sign.c:630]: nit: space before {

**Review by oreparaz — APPROVED** [review]

File: core/src/sign.c (hunk 2 above)
Comment (nmahendru) [reply core/src/sign.c:496]: actually I moved everything up there to facilitate the do while pattern. I think we can move it back down there.

Comment (nmahendru) [reply core/src/sign.c:497]: Actually that public key is inside two for loops and the cleanup is outside all of them.  I would just keep it at the top.

File: core/src/sign.c (hunk 3 above)
Comment (nmahendru) [reply core/src/sign.c]: I was thinking about running clang format on the whole file. thoughts ?

**Review by syncom — APPROVED** [review]
LGTM!

**Pull request merged**